Failure to build on current Debian

Hi, I'm maintaining pktanon in Debian and I was recently made aware that pktanon fails to build correctly on the latest Debian. I'm no too intimately familiar with the finer workings of C++, so I am forwarding the report to you.
Here's the log of the build failure:
```
g++ -DPACKAGE_NAME=\"pktanon\" -DPACKAGE_TARNAME=\"pktanon\" -DPACKAGE_VERSION=\"2.0-beta\" -DPACKAGE_STRING=\"pktanon\ 2.0-beta\" -DPACKAGE_BUGREPORT=\"bless@kit.edu\" -DPACKAGE_URL=\"\" -DPACKAGE=\"pktanon\" -DVERSION=\"2.0-beta\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_STDDEF_H=1 -DHAVE_CXX11=1 -DHAVE_SOCKET=1 -DHAVE_SYS_SOCKET_H=1 -DHAVE_ARPA_INET_H=1 -DHAVE_SYS_IOCTL_H=1 -DHAVE_NETINET_IN_H=1 -DHAVE_LINUX_IF_ETHER_H=1 -DHAVE_NETPACKET_PACKET_H=1 -DHAVE_NET_IF_H=1 -DHAVE_SYS_IOCTL_H=1 -DHAVE_FCNTL_H=1 -DHAVE_PCAP_PCAP_H=1 -I.  -O2 -pipe -Werror=return-type -I../include -g -DTRACE_ENABLED -DHAVE_LIBPCAP -I../libpktanon -Wdate-time -D_FORTIFY_SOURCE=2  -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -c -o OstreamOutput.o OstreamOutput.cpp
In file included from RawSocketRecordsHandler.cpp:20:
/usr/include/net/if.h:90:21: error: field ‘ifa_addr’ has incomplete type ‘sockaddr’
   90 |     struct sockaddr ifa_addr; /* Address of interface.  */
      |                     ^~~~~~~~
/usr/include/net/if.h:90:12: note: forward declaration of ‘struct sockaddr’
   90 |     struct sockaddr ifa_addr; /* Address of interface.  */
      |            ^~~~~~~~
/usr/include/net/if.h:93:18: error: field ‘ifu_broadaddr’ has incomplete type ‘sockaddr’
   93 |  struct sockaddr ifu_broadaddr;
      |                  ^~~~~~~~~~~~~
/usr/include/net/if.h:90:12: note: forward declaration of ‘struct sockaddr’
   90 |     struct sockaddr ifa_addr; /* Address of interface.  */
      |            ^~~~~~~~
/usr/include/net/if.h:94:18: error: field ‘ifu_dstaddr’ has incomplete type ‘sockaddr’
   94 |  struct sockaddr ifu_dstaddr;
      |                  ^~~~~~~~~~~
/usr/include/net/if.h:90:12: note: forward declaration of ‘struct sockaddr’
   90 |     struct sockaddr ifa_addr; /* Address of interface.  */
      |            ^~~~~~~~
/usr/include/net/if.h:137:18: error: field ‘ifru_addr’ has incomplete type ‘sockaddr’
  137 |  struct sockaddr ifru_addr;
      |                  ^~~~~~~~~
/usr/include/net/if.h:90:12: note: forward declaration of ‘struct sockaddr’
   90 |     struct sockaddr ifa_addr; /* Address of interface.  */
      |            ^~~~~~~~
/usr/include/net/if.h:138:18: error: field ‘ifru_dstaddr’ has incomplete type ‘sockaddr’
  138 |  struct sockaddr ifru_dstaddr;
      |                  ^~~~~~~~~~~~
/usr/include/net/if.h:90:12: note: forward declaration of ‘struct sockaddr’
   90 |     struct sockaddr ifa_addr; /* Address of interface.  */
      |            ^~~~~~~~
/usr/include/net/if.h:139:18: error: field ‘ifru_broadaddr’ has incomplete type ‘sockaddr’
  139 |  struct sockaddr ifru_broadaddr;
      |                  ^~~~~~~~~~~~~~
/usr/include/net/if.h:90:12: note: forward declaration of ‘struct sockaddr’
   90 |     struct sockaddr ifa_addr; /* Address of interface.  */
      |            ^~~~~~~~
/usr/include/net/if.h:140:18: error: field ‘ifru_netmask’ has incomplete type ‘sockaddr’
  140 |  struct sockaddr ifru_netmask;
      |                  ^~~~~~~~~~~~
/usr/include/net/if.h:90:12: note: forward declaration of ‘struct sockaddr’
   90 |     struct sockaddr ifa_addr; /* Address of interface.  */
      |            ^~~~~~~~
/usr/include/net/if.h:141:18: error: field ‘ifru_hwaddr’ has incomplete type ‘sockaddr’
  141 |  struct sockaddr ifru_hwaddr;
      |                  ^~~~~~~~~~~
/usr/include/net/if.h:90:12: note: forward declaration of ‘struct sockaddr’
   90 |     struct sockaddr ifa_addr; /* Address of interface.  */
      |            ^~~~~~~~
RawSocketRecordsHandler.cpp: In constructor ‘pktanon::RawSocketRecordsHandler::RawSocketRecordsHandler(pktanon::Stats&)’:
RawSocketRecordsHandler.cpp:52:25: error: ‘::socket’ has not been declared; did you mean ‘pktanon::socket’?
   52 |   socket_descriptor = ::socket ( AF_PACKET , SOCK_RAW , htons ( ETH_P_ALL ) );
      |                         ^~~~~~
      |                         pktanon::socket
In file included from /usr/include/pcap/socket.h:67,
                 from /usr/include/pcap/pcap.h:87,
                 from /usr/include/pcap.h:43,
                 from Utils.h:38,
                 from RawSocketRecordsHandler.cpp:14:
/usr/include/x86_64-linux-gnu/sys/socket.h:102:12: note: ‘pktanon::socket’ declared here
  102 | extern int socket (int __domain, int __type, int __protocol) __THROW;
      |            ^~~~~~
RawSocketRecordsHandler.cpp:75:34: error: cannot convert ‘sockaddr*’ to ‘const pktanon::sockaddr*’
   75 |   if ( bind ( socket_descriptor, ( struct ::sockaddr* ) &bind_addr,  sizeof ( struct sockaddr_ll ) ) < 0 )
      |                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                  |
      |                                  sockaddr*
In file included from /usr/include/pcap/socket.h:67,
                 from /usr/include/pcap/pcap.h:87,
                 from /usr/include/pcap.h:43,
                 from Utils.h:38,
                 from RawSocketRecordsHandler.cpp:14:
/usr/include/x86_64-linux-gnu/sys/socket.h:112:49: note:   initializing argument 2 of ‘int pktanon::bind(int, const pktanon::sockaddr*, pktanon::socklen_t)’
  112 | extern int bind (int __fd, __CONST_SOCKADDR_ARG __addr, socklen_t __len)
      |                                                 ^
In file included from RawSocketRecordsHandler.cpp:20:
/usr/include/net/if.h:90:12: note: class type ‘sockaddr’ is incomplete
   90 |     struct sockaddr ifa_addr; /* Address of interface.  */
      |            ^~~~~~~~
RawSocketRecordsHandler.cpp: In member function ‘virtual void pktanon::RawSocketRecordsHandler::read_packets()’:
RawSocketRecordsHandler.cpp:119:95: error: cannot convert ‘sockaddr*’ to ‘pktanon::sockaddr*’
  119 |     auto rcv_bytes_count = recvfrom ( socket_descriptor, original_packet, snaplen, MSG_TRUNC, ( struct ::sockaddr* ) &sa, &sa_len );
      |                                                                                               ^~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                                                                               |
      |                                                                                               sockaddr*
In file included from /usr/include/x86_64-linux-gnu/sys/socket.h:269,
                 from /usr/include/pcap/socket.h:67,
                 from /usr/include/pcap/pcap.h:87,
                 from /usr/include/pcap.h:43,
                 from Utils.h:38,
                 from RawSocketRecordsHandler.cpp:14:
/usr/include/x86_64-linux-gnu/bits/socket2.h:65:19: note:   initializing argument 5 of ‘ssize_t pktanon::recvfrom(int, void*, std::size_t, int, pktanon::sockaddr*, pktanon::socklen_t*)’
   65 |    __SOCKADDR_ARG __addr, socklen_t *__restrict __addr_len)
      |                   ^
In file included from RawSocketRecordsHandler.cpp:20:
/usr/include/net/if.h:90:12: note: class type ‘sockaddr’ is incomplete
   90 |     struct sockaddr ifa_addr; /* Address of interface.  */
      |            ^~~~~~~~
g++ -DPACKAGE_NAME=\"pktanon\" -DPACKAGE_TARNAME=\"pktanon\" -DPACKAGE_VERSION=\"2.0-beta\" -DPACKAGE_STRING=\"pktanon\ 2.0-beta\" -DPACKAGE_BUGREPORT=\"bless@kit.edu\" -DPACKAGE_URL=\"\" -DPACKAGE=\"pktanon\" -DVERSION=\"2.0-beta\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_STDDEF_H=1 -DHAVE_CXX11=1 -DHAVE_SOCKET=1 -DHAVE_SYS_SOCKET_H=1 -DHAVE_ARPA_INET_H=1 -DHAVE_SYS_IOCTL_H=1 -DHAVE_NETINET_IN_H=1 -DHAVE_LINUX_IF_ETHER_H=1 -DHAVE_NETPACKET_PACKET_H=1 -DHAVE_NET_IF_H=1 -DHAVE_SYS_IOCTL_H=1 -DHAVE_FCNTL_H=1 -DHAVE_PCAP_PCAP_H=1 -I.  -O2 -pipe -Werror=return-type -I../include -g -DTRACE_ENABLED -DHAVE_LIBPCAP -I../libpktanon -Wdate-time -D_FORTIFY_SOURCE=2  -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -c -o SocketOutput.o SocketOutput.cpp
g++ -DPACKAGE_NAME=\"pktanon\" -DPACKAGE_TARNAME=\"pktanon\" -DPACKAGE_VERSION=\"2.0-beta\" -DPACKAGE_STRING=\"pktanon\ 2.0-beta\" -DPACKAGE_BUGREPORT=\"bless@kit.edu\" -DPACKAGE_URL=\"\" -DPACKAGE=\"pktanon\" -DVERSION=\"2.0-beta\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_STDDEF_H=1 -DHAVE_CXX11=1 -DHAVE_SOCKET=1 -DHAVE_SYS_SOCKET_H=1 -DHAVE_ARPA_INET_H=1 -DHAVE_SYS_IOCTL_H=1 -DHAVE_NETINET_IN_H=1 -DHAVE_LINUX_IF_ETHER_H=1 -DHAVE_NETPACKET_PACKET_H=1 -DHAVE_NET_IF_H=1 -DHAVE_SYS_IOCTL_H=1 -DHAVE_FCNTL_H=1 -DHAVE_PCAP_PCAP_H=1 -I.  -O2 -pipe -Werror=return-type -I../include -g -DTRACE_ENABLED -DHAVE_LIBPCAP -I../libpktanon -Wdate-time -D_FORTIFY_SOURCE=2  -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -c -o LibpcapRecordsHandler.o LibpcapRecordsHandler.cpp
make[2]: *** [Makefile:446: RawSocketRecordsHandler.o] Error 1
```

You can also see the complete build log (e.g. for amd64) here:
https://buildd.debian.org/status/fetch.php?pkg=pktanon&arch=amd64&ver=2%7Egit20160407.0.2bde4f2%2Bdfsg-5%2Bb1&stamp=1572075234&raw=0

--- a/src/Utils.h
+++ b/src/Utils.h
@@ -13,6 +13,11 @@
 #include <PktAnon.h>
 // #include <transformations/ErrorCodes.h>
 
+# ifdef HAVE_LIBPCAP
+
+# include <pcap.h>
+# endif
+
 namespace pktanon
 {
 
@@ -34,8 +39,6 @@ inline void transform_packet ( PCAP_REC_HEADER& record_header, const uint8_t* or
 }
 
 # ifdef HAVE_LIBPCAP
-
-# include <pcap.h>
 
 inline void transform_packet ( struct pcap_pkthdr* pkt_header, const uint8_t* original_packet, uint8_t* transformed_packet, Stats& stats )
 {

